src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Windows.targets

```diff
@@ -46,6 +46,7 @@ The .NET Foundation licenses this file to you under the MIT license.
       <NativeLibrary Condition="'$(_targetArchitecture)' == 'x64'" Include="$(IlcSdkPath)$(VxSortSupportName)$(LibrarySuffix)" />
       <NativeLibrary Condition="'$(IlcMultiModule)' == 'true'" Include="$(SharedLibrary)" />
       <NativeLibrary Include="$(IlcSdkPath)$(StandaloneGCSupportName)$(LibrarySuffix)" />
+      <NativeLibrary Include="$(IlcSdkPath)zlibstatic$(LibFileExt)" />
```

Comment:
This needs to be `LibrarySuffix` and the library needs to be built in both regular and GuardCF flavors.

I think it would be easier if you just included the source in System.IO.Compression.Native like we do it for Brotli.

Reply:
Including the source means we need to replicate all of the logic in zlib-ng's CMake scripts to detect and support all of the hardware intrinsics (the majority of the reason to use zlib-ng in the first place).

Isn't it possible (though not performant) to build zlib-ng with `/guard:cf` in all scenarios (even though it wouldn't be useful in the non-GuardCF scenario?)

Reply:
I do not think that mixing and matching guard and non-guard binaries works well.

Reply:
(happy to be proven wrong)

Reply:
The docs say it works, but I'll test it locally to make sure. I wouldn't be amazed if the docs are wrong.

Reply:
From the [docs](https://learn.microsoft.com/en-us/cpp/build/reference/guard-enable-control-flow-guard?view=msvc-170):

> Code compiled by using /guard:cf can be linked to libraries and object files that aren't compiled by using the option. Only this code, when also linked by using the /guard:cf option and run on a CFG-aware operating system, has CFG protection. Because code compiled without the option won't stop an attack, we recommend that you use the option on all the code you compile. There's a small runtime cost for CFG checks, but the compiler analysis attempts to optimize away the checks on indirect jumps that can be proven to be safe.

I'm still validating it, but first few tests seem to work. Need to validate the compression tests in NativeAOT specifically. I'll report back.

Reply:
I've validated locally that we can enable `/guard:cf` unconditionally for zlib-ng and still link it into a non-GuardCF NativeAOT app and all of the compression tests pass. I'll push up a fix for this.